src/components/layout/AppBar.tsx

```diff
@@ -365,18 +397,58 @@ const WalletButton: FC<{}> = () => {
 export const AppBar: FC = () => {
   const accountOpen = useAccountOpen();
   const closeAccount = useCloseAccount();
+  const selectedMasset = useSelectedMasset();
   const { pathname } = useLocation();
   const home = pathname === '/';
+  const history = useHistory();
+  const selectMasset = useSetSelectedMasset();
+
+  const handleMusdClick = useCallback(() => {
+    selectMasset('mUSD');
+    history.push('/musd/mint');
+  }, [history, selectMasset]);
+
+  const handleMbtcClick = useCallback(() => {
+    selectMasset('mBTC');
+    history.push('/mbtc/mint');
+  }, [history, selectMasset]);
 
   return (
-    <Container inverted={accountOpen} home={home}>
+    <Container inverted={accountOpen}>
       <Inner>
         <Top>
-          <Logo inverted={accountOpen} full={home}>
+          <Logo inverted={accountOpen}>
             <Link to="/" title="Home" onClick={closeAccount}>
               <LogoSvg />
             </Link>
           </Logo>
+          {home
+            ? null
+            : selectedMasset && (
+                <MassetSwitch
+                  onClick={
+                    selectedMasset.name === 'mUSD'
+                      ? handleMbtcClick
+                      : handleMusdClick
+                  }
+                >
+                  <span>{selectedMasset.name}</span>
+                  <IconContainer name={selectedMasset.name}>
+                    {selectedMasset.name === 'mUSD' ? (
+                      <MusdIcon />
```

Comment:
Perhaps this would be simpler if there were an `icon` prop in the state for mAssets